src/main/java/frc/robot/subsystems/drivetrain/BaseSwerveSubsystem.java

```diff
@@ -221,13 +220,44 @@ public void applyLock() {
      * @param relative Whether to use relative powers instead of field-oriented control.
      */
     public void setDrivePowers(double xPower, double yPower, double angularPower, boolean relative) {
+        // Increment target heading from angular power
+        this.targetHeading = targetHeading.plus(new Rotation2d(MAX_OMEGA * angularPower));
```

Comment:
this essentially is saying that our target heading when full turn power is applied should be `MAX_OMEGA` away. this is an issue because `MAX_OMEGA` is rad/s but each loop timing is like 20ms.